src/main/java/net/neoforged/neoforge/client/extensions/IRenderStateExtension.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.extensions;
+
+import net.minecraft.util.context.ContextKey;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Extension class for render state objects. Implemented by {@link net.neoforged.neoforge.client.BaseRenderState} for
+ * simple class extension.
+ */
+public interface IRenderStateExtension {
```

Comment:
Is there any value in keeping this interface around instead of folding it into `BaseRenderState`?

Reply:
Fair enough